juju/framework.py

```diff
@@ -541,6 +557,25 @@ def snapshot(self):
     def restore(self, snapshot):
         self._cache = snapshot
 
+    def clear_dirty(self):
+        self._dirty = False
+
+    def set_dirty(self):
+        self._dirty = True
+
+    def is_dirty(self):
+        """Check if a copy of a stored state data is dirty in memory."""
+        try:
+            dirty = getattr(self, '_dirty')
+        except AttributeError:
+            # When StoredStateData is restored from persistent storage its __init__ is not called so it is clear that it is not dirty yet.
+            dirty = False
+        return dirty
```

Comment:
That feels like a very complex implementation of a boolean. Why isn't this the more straightforward `data.dirty = True/False`?